pelican/__init__.py

```diff
@@ -410,73 +500,20 @@ def main():
                 [''],
                 pelican.ignore_files)
 
-        if args.autoreload:
+        if args.autoreload and args.listen:
+            autoreload_thread = threading.Thread(
+                target=autoreload, args=(watchers, args, old_static))
+            listen_thread = threading.Thread(
+                target=listen,
+                args=(args.bind, args.port, settings.get("OUTPUT_PATH")))
+            autoreload_thread.start()
+            listen_thread.start()
```

Comment:
Well... Did you test this? :)

Couple issues here. 
- Cannot quit this (used to be `^C` aka `KeyboardInterrupt`),  you need to kill the process.
- And exceptions inside threads won't propagate back to main caller. i.e. an exception in `--autoreload` mode with `--debug` is supposed to stop pelican (see line 454) but won't. Again, it hangs.